### PR TITLE
Add NCI download DAGs for WOfS and Geomedian summary products

### PR DESCRIPTION
**Pull request by jmettes**
Add DAG steps for downloading summary products for WOfS 

* ga_ls_wo_fq_apr_oct_3
* ga_ls_wo_fq_cyear_3
* ga_ls_wo_fq_myear_3
* ga_ls_wo_fq_nov_mar_3

and Geomedian

* ga_ls5t_nbart_gm_cyear_3
* ga_ls7e_nbart_gm_cyear_3
* ga_ls8c_nbart_gm_cyear_3

### Cost

I don't think syncing these directories daily is a significant cost. 

The pricing is [$0.0055](https://aws.amazon.com/s3/pricing/) per 1,000 S3 `ListObjects`  requests. Each `LIST` request returns a [maximum of 1,000 keys](https://forums.aws.amazon.com/thread.jspa?threadID=268525).

Taking a look at the WOfS data:

```
[jm5638@gadi-login-06 ga_ls_wo_3]$ cd /g/data/jw04/ga/ga_ls_wo_3
[jm5638@gadi-login-06 ga_ls_wo_3]$ find . -type f | wc -l
2378910
```

That would be about ~$0.013 a day to list all of `ga_ls_wo_3`.

### CODE REVIEW

File: dags/collection3/nci_c3_download_derivs.py [hunk 1]
```diff
@@ -31,93 +31,48 @@
...
-            COMMON
-            + """
-            module load dea
-            cd /g/data/v10/work/c3_download_derivs/{{ts_nodash}}
-
-            awk '/odc-metadata.yaml/ {print "/g/data/jw04/ga/ga_ls_wo_3/" $3}' ga_ls_wo_3.download.log  | \
-            xargs -P 4 datacube -v dataset add --no-verify-lineage --product ga_ls_wo_3
-        """
-        ),
-        # Attempt to index downloaded datasets, even if there were some failures in the download
```
Comment (omad) [inline dags/collection3/nci_c3_download_derivs.py:100]: This `trigger_rule` and comment are important. (and if it's not clear we should improve it).

We've had trouble with data being downloaded but not indexed, when s5cmd only partially completes. (eg. downloads 10 of 15 datasets). The Airflow task is marked as *failed*, and without this `trigger_rule` change the indexing job doesn't run.

If the download task runs a second time, it doesn't download the data that's already been downloaded, and so aren't included in the log file that we index from.

File: dags/collection3/nci_c3_download_derivs.py [hunk 2]
```diff
@@ -31,93 +31,48 @@
...
+            "ga_ls_wo_3",
+            "ga_ls_wo_fq_apr_oct_3",
+            "ga_ls_wo_fq_nov_mar_3",
+            "ga_ls_wo_fq_cyear_3",
+            "ga_ls_wo_fq_myear_3",
 
-    # Sync WOfS directory using a Gadi Data Mover node
-    sync_wofs = SSHOperator(
-        task_id="sync_wofs",
-        # Run on the Gadi Data Mover node, it's specifically spec'd for data transfers, and
```
Comment (omad) [inline dags/collection3/nci_c3_download_derivs.py:60]: This comment about using a Gadi Data Mover node for the transfers is relevant and should be kept.

File: dags/collection3/nci_c3_download_derivs.py [hunk 3]
```diff
@@ -31,93 +31,48 @@
...
-        # we use s5cmd to transfer using lots of threads to max out the network pipe, and quickly
-        # walk both the S3 tree and the Lustre FS tree.
-        remote_host="gadi-dm.nci.org.au",
-        # There have been random READ failures when performing this download. So retry a few times.
```
Comment (omad) [inline dags/collection3/nci_c3_download_derivs.py:64]: I'm not sure about this comment, maybe it's truth, maybe it's lies...

**Review by omad — CHANGES_REQUESTED** [review]
Please add the `trigger_rule` and comment (or improved commentary) back in, otherwise looks great.

File: dags/collection3/nci_c3_download_derivs.py (hunk 3 above)
Comment (jmettes) [reply dags/collection3/nci_c3_download_derivs.py:64]: I'll leave it in for now - seems to at least serve as a warning of potentially being flaky...

**Review by emmaai — APPROVED** [review]
💯

**Review by omad — APPROVED** [review]

**Pull request merged**